src/BaseNbtSerializer.php

```diff
@@ -52,12 +51,9 @@ public function __construct(){
 	 */
 	private function readRoot(int $maxDepth) : TreeRoot{
 		$type = $this->readByte();
-		if($type !== NBT::TAG_Compound){
-			throw new NbtDataException("Expected TAG_Compound at the start of buffer");
-		}
 
 		$rootName = $this->readString();
-		return new TreeRoot(CompoundTag::read($this, new ReaderTracker($maxDepth)), $rootName);
+		return new TreeRoot(NBT::createTag($type, $this, new ReaderTracker($maxDepth)), $rootName);
```

Comment:
This will cause [this](https://github.com/pmmp/NBT/blob/master/src/NBT.php#L90) to be thrown when `TAG_End` is encountered. As per the code on 0.2, there should be an explicit check for `TAG_End`, otherwise it will misleadingly say that tag type `0` is unknown.